packages/react-instantsearch/src/widgets/__tests__/FrequentlyBoughtTogether.test.tsx

```diff
@@ -81,4 +82,149 @@ describe('FrequentlyBoughtTogether', () => {
     expect(root).toHaveClass('MyFrequentlyBoughtTogether', 'ROOT');
     expect(root).toHaveAttribute('aria-hidden', 'true');
   });
+
+  test('renders custom layout component', async () => {
+    const client = createRecommendSearchClient({
+      minimal: true,
+    });
+    const { container } = render(
+      <InstantSearchTestWrapper searchClient={client}>
+        <FrequentlyBoughtTogether
+          objectIDs={['1']}
+          layoutComponent={({ items }) => (
+            <ul>
+              {items.map((item) => (
+                <li key={item.objectID}>
+                  <p>{item.objectID}</p>
+                </li>
+              ))}
+            </ul>
+          )}
+        />
+      </InstantSearchTestWrapper>
+    );
+
+    await waitFor(() => {
+      expect(client.getRecommendations).toHaveBeenCalledTimes(1);
+    });
+
+    await waitFor(() => {
+      expect(container.querySelector('.ais-FrequentlyBoughtTogether'))
+        .toMatchInlineSnapshot(`
+        <section
+          class="ais-FrequentlyBoughtTogether"
+        >
+          <h3
+            class="ais-FrequentlyBoughtTogether-title"
+          >
+            Frequently bought together
+          </h3>
+          <ul>
+            <li>
+              <p>
+                1
+              </p>
+            </li>
+            <li>
+              <p>
+                2
+              </p>
+            </li>
+          </ul>
+        </section>
+        `);
+    });
+  });
+
+  test('renders Carousel as a layout component', async () => {
+    const client = createRecommendSearchClient({
+      minimal: true,
+    });
+    const { container } = render(
+      <InstantSearchTestWrapper searchClient={client}>
+        <FrequentlyBoughtTogether
+          objectIDs={['1']}
+          itemComponent={({ item }) => <p>{item.objectID}</p>}
+          layoutComponent={(props) => (
+            <Carousel
+              {...props}
+              previousIconComponent={() => <p>Previous</p>}
```

Comment:
Can you also test passing custom `classNames` and `translations` since they're available APIs?

Same for the other widgets.